pkg/egressgateway/manager.go

```diff
@@ -116,16 +113,16 @@ type Manager struct {
 	// endpoints allows reading endpoint CRD from k8s.
 	endpoints resource.Resource[*k8sTypes.CiliumEndpoint]
 
+	// endpointStore is a read-only store of CiliumEndpoint resources
+	endpointStore resource.Store[*k8sTypes.CiliumEndpoint]
```

Comment:
A meta comment: Why are we changing the handling of `CiliumEndpoint` at all? It's wrong to even use `CiliumEndpoint` here as it doesn't work for kvstore mode in which the whole use of the `CiliumEndpoint` CRD should be disabled! The whole point of the kvstore mode is to make Cilium more scalable by disseminating the data in an efficient form via etcd.

Instead the use of it here should be removed completely and replaced with ipcache subscription.